internal/pkg/client/oci/pull.go

```diff
@@ -108,6 +109,9 @@ func PullToFile(ctx context.Context, imgCache *cache.Handle, pullTo, pullFrom, t
 
 	src, err := pull(ctx, imgCache, directTo, pullFrom, tmpDir, ociAuth, noHTTPS, noCleanUp)
 	if err != nil {
+		if strings.Contains(err.Error(), "unsupported image-specific operation on artifact with type \"application/vnd.unknown.config.v1+json\"") {
+			return "", fmt.Errorf("while building SIF from layers: %v, you might need to change the protocol to oras://", err)
```

Comment:
I think the "while building SIF from layers" is supposed to be added at a higher level, not in this function.  Change it to just "%v; try changing the protocol to oras://".

Reply:
thanks for the comment! reflected in this commit https://github.com/apptainer/apptainer/pull/703/commits/ed8c9af19331d20eb636b00bb05d4a7517f03ac8